pkg/vault/secrets.go

```diff
@@ -11,17 +10,17 @@ func (v *Vault) GetSecretKey(path string, key string) (string, error) {
 
 	secret, err := v.client.Logical().Read(path)
 	if err != nil {
-		return "", err
+		return "", v.parseError(err)
```

Comment:
I would assume you want to return Error (or whatever we might change it too) so if someone wants to handle it a different way farther down (say another stimpack calling it) they would not have to see if its castable.